src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfCodec.java

```diff
@@ -0,0 +1,376 @@
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * {@link htsjdk.tribble.Tribble} Codec to read data from a GENCODE GTF file.
+ *
+ * GENCODE GTF Files are defined here: https://www.gencodegenes.org/data_format.html
+ *
+ * Created by jonn on 7/21/17.
+ */
+final public class GencodeGtfCodec extends AbstractFeatureCodec<GencodeGtfFeature, LineIterator> {
+
+    protected final Logger logger = LogManager.getLogger(this.getClass());
+
+    private static final String COMMENT_START = "##";
+
+    static final int NUM_COLUMNS = 9;
+
+    private long currentLineNum = 1;
+    protected List<String> header = new ArrayList<>();
+
+    // ============================================================================================================
+
+    public GencodeGtfCodec() {
+        super(GencodeGtfFeature.class);
+    }
+
+    // ============================================================================================================
+    // Trivial override methods that are pulled form AsciiFeatureCodec
+    // This was done to ensure that this was a reasonable Codec class (with good interfaces for reading features).
+
+    @Override
+    public void close(final LineIterator lineIterator) {
+        CloserUtil.close(lineIterator);
+    }
+
+    @Override
+    public boolean isDone(final LineIterator lineIterator) {
+        return !lineIterator.hasNext();
+    }
+
+    @Override
+    public LineIterator makeSourceFromStream(final InputStream bufferedInputStream) {
+        return new LineIteratorImpl(new SynchronousLineReader(bufferedInputStream));
+    }
+
+    @Override
+    public FeatureCodecHeader readHeader(final LineIterator lineIterator) throws IOException {
+        return new FeatureCodecHeader(readActualHeader(lineIterator), FeatureCodecHeader.NO_HEADER_END);
+    }
+
+    @Override
+    public LocationAware makeIndexableSourceFromStream(final InputStream bufferedInputStream) {
+        final PositionalBufferedStream pbs;
+        if (bufferedInputStream instanceof PositionalBufferedStream) {
+            pbs = (PositionalBufferedStream) bufferedInputStream;
+        } else {
+            pbs = new PositionalBufferedStream(bufferedInputStream);
+        }
+        return new AsciiLineReaderIterator(new AsciiLineReader(pbs));
+    }
+
+    // ============================================================================================================
+
+    @Override
+    public GencodeGtfFeature decode(final LineIterator lineIterator) {
+
+        GencodeGtfFeature decodedFeature = null;
+
+        // Create some caches for our data (as we need to group it):
+        GencodeGtfGeneFeature gene = null;
+        GencodeGtfTranscriptFeature transcript = null;
+        final ArrayList< GencodeGtfExonFeature > exonStore = new ArrayList<>();
+        final ArrayList< GencodeGtfFeature > leafFeatureStore = new ArrayList<>();
```

Comment:
I'm pretty sure I'm not using anything ArrayList-specific.  I'll update them to be Lists.

Sounds good. Yeah... Old habits die hard.  I've spent a lot of time in C++ land... Sometimes my hands just put the spaces in.